infrastructure/data_access/http/synthetics_repo.py

```diff
@@ -112,17 +144,16 @@ def scale_to_percents(val: str) -> MetricValue:
     return MetricValue(float(val) * 100.0)
 
 
-def make_internal_agents(agents, agent_ids) -> Agents:
+def transform_to_internal_agents(agents) -> Agents:
```

Comment:
I always wondered why the Agents class cannot have a constructor (or a class method) allowing to create an instance based on the API response. Agents are never constructed in any other way (AFAICT).

Reply:
There should be no dependency going from domain to infrastructure/presentation; this makes the core concepts independent from "outer world" eg you dont need to modify the domain if you decide to move from http -> grpc or you change presentation style. Domain package provides basic building blocks that other packages build on top of

Reply:
I agree that there should be no direct dependency between infrastructure and presentation (other than the infrastructure providing the data required by the presentation). My suggestion does not create any dependency of this kind. I also agree that it is good engineering to keep building blocks independent and reusable. In this case, I'm pretty sure that both REST and gRPC responses will actually provide identical dictionaries, so one constructor would work for both. I'm though fine with keeping construction of Agents as is. We have more important problems to work on :).